test/Microsoft.AspNetCore.SignalR.Redis.Tests/RedisHubLifetimeManagerTests.cs

```diff
@@ -1,558 +1,58 @@
 
-        [Fact]
-        public async Task InvokeConnectionAsyncOnNonExistentConnectionDoesNotThrow()
+        public override HubLifetimeManager<MyHub> CreateNewHubLifetimeManager(TestRedisServer backplane)
         {
-            var server = new TestRedisServer();
-
-            var manager = CreateLifetimeManager(server);
-            await manager.SendConnectionAsync("NotARealConnectionId", "Hello", new object[] { "World" }).OrTimeout();
+            return CreateLifetimeManager(backplane);
         }
 
-        [Fact]
-        public async Task InvokeAllAsyncWithMultipleServersWritesToAllConnectionsOutput()
+        public override TestRedisServer CreateBackplane()
         {
-            var server = new TestRedisServer();
-
-            var manager1 = CreateLifetimeManager(server);
-            var manager2 = CreateLifetimeManager(server);
-
-            using (var client1 = new TestClient())
-            using (var client2 = new TestClient())
-            {
-                var connection1 = HubConnectionContextUtils.Create(client1.Connection);
-                var connection2 = HubConnectionContextUtils.Create(client2.Connection);
-
-                await manager1.OnConnectedAsync(connection1).OrTimeout();
-                await manager2.OnConnectedAsync(connection2).OrTimeout();
-
-                await manager1.SendAllAsync("Hello", new object[] { "World" }).OrTimeout();
-
-                await AssertMessageAsync(client1);
-                await AssertMessageAsync(client2);
-            }
+            return new TestRedisServer();
         }
 
-        [Fact]
-        public async Task InvokeAllAsyncWithMultipleServersDoesNotWriteToDisconnectedConnectionsOutput()
+        public override HubLifetimeManager<MyHub> CreateNewHubLifetimeManager()
         {
-            var server = new TestRedisServer();
-
-            var manager1 = CreateLifetimeManager(server);
-            var manager2 = CreateLifetimeManager(server);
-
-            using (var client1 = new TestClient())
-            using (var client2 = new TestClient())
-            {
-                var connection1 = HubConnectionContextUtils.Create(client1.Connection);
-                var connection2 = HubConnectionContextUtils.Create(client2.Connection);
+            _server = new TestRedisServer();
 
-                await manager1.OnConnectedAsync(connection1).OrTimeout();
-                await manager2.OnConnectedAsync(connection2).OrTimeout();
-
-                await manager2.OnDisconnectedAsync(connection2).OrTimeout();
-
-                await manager2.SendAllAsync("Hello", new object[] { "World" }).OrTimeout();
-
-                await AssertMessageAsync(client1);
-
-                Assert.Null(client2.TryRead());
-            }
+            return CreateLifetimeManager(_server);
         }
 
-        [Fact]
-        public async Task InvokeConnectionAsyncOnServerWithoutConnectionWritesOutputToConnection()
+        class TestObject
```

Comment:
Is this still needed?

Reply:
It's used in one of the tests. I didn't really want to change the inner workings of the tests